crates/rqbit/src/main.rs

```diff
@@ -908,9 +946,17 @@ async fn start_http_api(
         Some(log_config.rust_log_reload_tx),
         Some(log_config.line_broadcast),
     );
+    let systemd_listener = systemd_socket_activation().unwrap_or_else(|e| {
```

Comment:
I'm happy to pass the socket into `start_http_api` instead, that'll just require a little more refactoring.